src/game.rs

```diff
@@ -37,6 +39,8 @@ pub enum GameResult {
 pub struct Game {
     start_pos: Board,
     moves: Vec<Action>,
+    pgn: String,
+    move_number: usize,
```

Comment:
both start_pos and all moves are in `Game`, so it doesn't look like it also always needs to store the PGN: if people use this Game struct for anything else, this is extra useless cost.
These should probably just be local variables of the Display implementation.